### DISCUSSION TITLE
【TypeScript版本】有没有想法改成TS

### DISCUSSION THREAD

**Discussion by JensonMiao** [tangly1024/NotionNext#2006] [Ideas]

**Comment by tangly1024** [comment]
有想过，TS会带来更强的生产力，TS的版本将是一个大版本的更新

然而我有些担心的是，全面TS后，使用NotionNext的用户会在更新和使用上遇到一点困难：
1. 很多用户喜欢自己调整页面的样式，全面TS改动后，更新新版本时难免会遇到大量合并冲突的问题。
2. 后续用户需要自定义时，需要再付出学习ts语法的成本

**Comment by qfdk** [comment]
并不是很喜欢TS，不适用TS 写起来更加灵活：我们团队之前还在是不是使用TS上面做过讨论，还是JS 比较方便。

**Comment by Vixcity** [comment]
TS 非常的不方便，本身 JS 一个特性就是弱类型，现在在给他加上一个强类型，不仅增加了学习成本，还一不小心就类型有误，非常难受
